illumina_somaticVariants.pm

```diff
@@ -34,8 +34,13 @@ sub parseSamples {
         if($opt{SET_HAS_METADATA} eq "yes"){
             print "Reading Tumour/Reference names from metadata\n\n";
             my $metadata = metadataParse($opt{OUTPUT_DIR});
-            push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
-            push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            if($sample =~ $metadata->{'ref_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
+            } elsif ($sample =~ $metadata->{'tumor_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            } else {
+              print "WARNING: Somatic variants are set to yes but metadata does not match sample names";
```

Comment:
For clarity, maybe something like `"Metadata is set to yes but \"$sample\" does not match a metadata sample name"`? What if the metadata file exists but does not contain the right keys?